flink-connector-debezium/src/main/java/com/alibaba/ververica/cdc/debezium/DebeziumSourceFunction.java

```diff
@@ -354,11 +364,17 @@ private void snapshotHistoryRecordsState() throws Exception {
 
         if (engineInstanceName != null) {
             schemaRecordsState.add(engineInstanceName);
-
-            Collection<SchemaRecord> records = StateUtils.retrieveHistory(engineInstanceName);
-            DocumentWriter writer = DocumentWriter.defaultWriter();
-            for (SchemaRecord record : records) {
-                schemaRecordsState.add(writer.write(record.toDocument()));
+            Collection<SchemaRecord> records = checkStatusAndRetrieveHistory(engineInstanceName);
+            if (records.isEmpty()) {
+                throw new FlinkRuntimeException(
```

Comment:
We shoudn't throw exception here, because history might be empty when the job is just started.